optaplanner-persistence/optaplanner-persistence-jpa/pom.xml

```diff
@@ -84,11 +84,85 @@
       <artifactId>h2</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.codehaus.btm</groupId>
+      <artifactId>btm</artifactId>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 
+  <properties>
+    <maven.btm.maxPoolSize>16</maven.btm.maxPoolSize>
+    <maven.hibernate.dialect>org.hibernate.dialect.H2Dialect</maven.hibernate.dialect>
+    <maven.datasource.classname>bitronix.tm.resource.jdbc.lrc.LrcXADataSource</maven.datasource.classname>
+    <maven.jdbc.driver.class>org.h2.Driver</maven.jdbc.driver.class>
+    <maven.jdbc.db.name/>
+    <maven.jdbc.db.port/>
+    <maven.jdbc.db.server/>
+    <maven.jdbc.driver.jar/>
+    <maven.jdbc.username></maven.jdbc.username>
+    <maven.jdbc.password></maven.jdbc.password>
+    <maven.jdbc.url>jdbc:h2:</maven.jdbc.url>
+    <maven.jdbc.schema>public</maven.jdbc.schema>
+  </properties>
+
+  <profiles>
+    <profile>
+      <id>database</id>
+      <dependencyManagement>
+        <dependencies>
+          <dependency>
+            <groupId>org.jbpm</groupId>
+            <artifactId>downloaded-jdbc-driver-jar</artifactId>
+            <version>1.0</version>
+          </dependency>
+        </dependencies>
+      </dependencyManagement>
+
+      <dependencies>
+        <!--  Add the downloaded jdbc driver jar for the database to the classpath.
+              !: <groupId>, <artifactId> and <version> are just placeholders and DO NOT MATTER! -->
+        <dependency>
+          <groupId>org.jbpm</groupId>
+          <artifactId>downloaded-jdbc-driver-jar</artifactId>
+          <scope>system</scope>
+          <systemPath>${maven.jdbc.driver.jar}</systemPath>
+        </dependency>
+      </dependencies>
```

Comment:
@psiroky This uses the same trick as the drools-persistence-jpa tests  and it's a bit special: the artifactId downloaded-jdbc-driver-jar doesn't actually exist: it gets replaced on the fly base don the jenkins job configuration's database type it's running against. There are many database types (oracle, mysql, ...).

Would switching to that surefire approach still allow that? If so, we need to do that both here and on the drools-persistence-jpa pom.